### PR TITLE
Feature/add config validation tests

### PR DESCRIPTION
**Pull request by SuperITMan**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [X] The commit message follows our guidelines: https://github.com/NationalBankBelgium/code-style/blob/master/CONTRIBUTING.md#-commit-message-guidelines
-   [X] Tests for the changes have been added (for bug fixes / features)
-   [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

There is no test in the repo.

The stylelint 9.10.x configuration extends incorrectly "stylelint-config-prettier".

Issue Number: #17 

## What is the new behavior?

Tests are now implemented to validate configurations of:
  - prettier
  - stylelint
  - tsconfig

It's not possible to validate tslint configurations. See: https://github.com/palantir/tslint/issues/889

Since tests are implemented, the stylelint 9.10.x configuration has been fixed.
But it means we have a breaking change here.
Should we release a major release after this or a minor would be enough ?

## Does this PR introduce a breaking change?

```
[X] Yes
[ ] No
```

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+do
+  travisFoldStart "validate tsconfig ${TSCONFIG_VERSION}" "no-xtrace"
+  logDebug "create temporary tsconfig file: /tmp/tsconfig-test-file-${TSCONFIG_VERSION}.json"
+  echo "{\"extends\":\"${tsconfigFolder}/${TSCONFIG_VERSION}/tsconfig.json\",\"compilerOptions\":{\"baseUrl\":\".\"}}" > /tmp/tsconfig-test-file-${TSCONFIG_VERSION}.json
+  logDebug "run command \"npx typescript@${TSCONFIG_VERSION} -p /tmp/tsconfig-test-file-${TSCONFIG_VERSION}.json\""
+  npx typescript@${TSCONFIG_VERSION} -p /tmp/tsconfig-test-file-${TSCONFIG_VERSION}.json
+  travisFoldEnd "validate tsconfig ${TSCONFIG_VERSION}"
+done
+
+#travisFoldEnd "validate tsconfig configurations" "no-xtrace"
```
Comment (christophercr) [inline test.sh]: Small typo `tsconfig` -> `tslint`

Comment (SuperITMan) [reply test.sh]: Good catch! Thanks :blush:

**Comment by SuperITMan** [comment]
@christophercr you're right, it was not documented and it was a mistake since the beginning.
It will be documented as a fix but not as a BREAKING CHANGE. I updated my PR to apply this :blush:

**Comment by christophercr** [comment]
@SuperITMan I suggest to change the commit message like this to clarify that it was a mistake and we never supported that, what do you think?

```
The `stylelint/9.10.x` config was mistakenly extending from `stylelint-config-prettier` and therefore the `"extends": "stylelint-config-prettier"` has been removed from that config. Please add that yourself in your stylelint config if necessary.
```

**Comment by SuperITMan** [comment]
@christophercr Actually I generated the CHANGELOG to see how it is displayed but our message does not appear... :confused: 

### Bug Fixes

* **stylelint:** remove wrong extend to "stylelint-config-prettier" in stylelint 9.10.x configuration ([b09eb47](https://github.com/NationalBankBelgium/code-style/commit/b09eb47)), closes [#17](https://github.com/NationalBankBelgium/code-style/issues/17)


Would you know if there is a way to add more information about a feature/fix ? :blush: 
Or maybe it is only for the breaking changes ?

**Comment by christophercr** [comment]
Mmm normally we follow the angular commit guidelines and [there](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#body) they say the `ISSUES CLOSED` should be in the footer so any other description should be before that.

Could you try that?

**Comment by SuperITMan** [comment]
I tried both but the result is the same...

**Comment by christophercr** [comment]
Btw the `test.sh` script fails on Windows:

```
$ npm run test

===> test.sh validate prettier configurations ==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>

===> test.sh validate prettier 1.16.x ==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! @nationalbankbelgium/code-style@1.0.1 test: `bash ./test.sh`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the @nationalbankbelgium/code-style@1.0.1 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

Any idea why?

**Comment by nicanac** [comment]
> Btw the `test.sh` script fails on Windows:
> 
> ```
> $ npm run test
> 
> ===> test.sh validate prettier configurations ==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>
> 
> ===> test.sh validate prettier 1.16.x ==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==
> npm ERR! code ELIFECYCLE
> npm ERR! errno 2
> npm ERR! @nationalbankbelgium/code-style@1.0.1 test: `bash ./test.sh`
> npm ERR! Exit status 2
> npm ERR!
> npm ERR! Failed at the @nationalbankbelgium/code-style@1.0.1 test script.
> npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> ```
> 
> Any idea why?

Isn't it an issue with an env variable.

**Comment by SuperITMan** [comment]
I understand what's happening...

When we execute the command
>  npx prettier@1.16.x --config /tmp/prettier-test-file-1.16.x.js --write /tmp/prettier-test.css

We get the following: 
```bash
npx: installato 1 in 2.098s
[error] Invalid configuration file: Cannot find module '/c/Users/superitman/DEV/code-style/lib/prettier/1.16.x/.prettierrc.js'
[error] Require stack:
[error] - C:\Users\superitman\AppData\Local\Temp\1\prettier-test-file-1.16.x.js
[error] - C:\Users\superitman\AppData\Roaming\npm-cache\_npx\8780\node_modules\prettier\third-party.js
[error] - C:\Users\superitman\AppData\Roaming\npm-cache\_npx\8780\node_modules\prettier\bin-prettier.js
```

It's not working because of the content in the tmp file we create in the test:

```bash
cat /tmp/prettier-test-file-1.16.x.js
module.exports=require('/c/Users/superitman/DEV/code-style/lib/prettier/1.16.x/.prettierrc.js');
```

If we change the path as follows, it will work:

```bash
cat /tmp/prettier-test-file-1.16.x.js
module.exports=require('C:\\Users\\superitman\\DEV\\code-style\\lib\\prettier\\1.16.x\\.prettierrc.js');
```

An easy fix would be to add the following code at the top of test.sh instead of `readonly currentDir=$(cd $(dirname $0); pwd)`

```bash
case "$(uname -s)" in
  CYGWIN*|MINGW32*|MSYS*|MINGW*)
    readonly currentDir=$(cd $(dirname $0); pwd -W)
    ;;
  *)
    readonly currentDir=$(cd $(dirname $0); pwd)
    ;;
esac
```

What do you think @christophercr @nicanac ?

**Comment by christophercr** [comment]
Mmm what if we do `cd /tmp` before running the `npx` command and remove the `/tmp/xxx` from the paths in that command?

**Comment by SuperITMan** [comment]
I implemented another way to create the temp files using a ".tmp" folder in code-style. 
What do you think @christophercr ?
I'll squash this commit after if you're ok 😊

**Comment by christophercr** [comment]
I confirm it is working now in Windows 👍

**Pull request merged**